Golem/Yagna/ProcessFactory.cs

```diff
@@ -60,77 +60,111 @@ public override void WriteLine()
+            }
+
+            foreach (var (k, v) in env)
+            {
+                if (startInfo.EnvironmentVariables.ContainsKey(k))
+                    startInfo.EnvironmentVariables[k] = v;
+                else
+                    startInfo.EnvironmentVariables.Add(k, v);
+            }
+
+            var process = new Process
+            {
+                StartInfo = startInfo,
+                EnableRaisingEvents = true,
+            };
+
+            return process.Start() ?
+                process : throw new GolemProcessException($"Failed to start Golem process: {process}");
         }
 
-        static Shell.Options updateOptions(Shell.Options options, string workDir, Dictionary<string, string> env)
+        private static void BindOutputEventHandlers(Process proc)
         {
-            options = options
-                .EnvironmentVariables(env)
-                .WorkingDirectory(workDir)
-                .ThrowOnError(true)
-                .DisposeOnExit(false)
-                .StartInfo(info =>
-                {
-                    info.CreateNoWindow = true;
-                });
-            return options;
+            proc.OutputDataReceived += OnOutputDataRecv;
+            proc.ErrorDataReceived += OnErrorDataRecv;
+            proc.BeginErrorReadLine();
+            proc.BeginOutputReadLine();
+        }
+
+        private static void OnOutputDataRecv(object sender, DataReceivedEventArgs e)
+        {
+            // _logger.LogInformation($"{e.Data}");
+            Console.WriteLine($">>> {e.Data}");
+        }
+        private static void OnErrorDataRecv(object sender, DataReceivedEventArgs e)
+        {
+            // _logger.LogInformation($"{e.Data}");
+            Console.WriteLine($">>> {e.Data}");
         }
 
         public static string BinName(string name)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                return Path.ChangeExtension(name, ".exe");
-            }
-            else
-            {
-                return Path.GetFileNameWithoutExtension(name);
-            }
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                Path.ChangeExtension(name, ".exe") : Path.GetFileNameWithoutExtension(name);
         }
 
         // TODO instead of passing logger as param and keeping `cmd` as variable, keep ProcessFactory (with `cmd` and `logger`) as a variable.
-        public static async Task<int> StopCmd(Command cmd, int stopTimeoutMs = 30_000, ILogger? logger = null)
+        public static async Task<int> StopProcess(Process proc, int stopTimeoutMs = 30_000, ILogger? logger = null)
         {
-            if (!cmd.Process.HasExited)
+            if (proc.HasExited)
+            {
+                logger?.LogWarning("Process has exited already.");
+                return proc.ExitCode;
+            }
+            if (Command.TryAttachToProcess(proc.Id, out var cmd))
```

Comment:
I use Medallion lib for sending signals on multiple OS only when stopping a process.
First it attaches to `Process`, then it tries to send a signal.
It is not ideal, because method accepts any `Process`, but it will work only with process with a Console (in our case these are `ya-provider run`, and `yagna service run` processes).